Octokit/Http/SimpleJsonSerializer.cs

```diff
@@ -92,6 +92,17 @@ public override object DeserializeObject(object value, Type type)
                 {
                     if (ReflectionUtils.GetTypeInfo(type).IsEnum)
                     {
+                        //first try to get all custom attributes
```

Comment:
Or we create a customAttribute class `DeserializeProperlyAttribute` for example. And then
```
public override object DeserializeObject(object value, Type type)
            {
                var stringValue = value as string;
                var jsonValue = value as JsonObject;
                if (stringValue != null)
                {
                    if (ReflectionUtils.GetTypeInfo(type).IsEnum)
                    {
                        if(Attribute.IsDefined(type, typeof(DeserializeProperlyAttribute)))
                        {
                            //first try to get all custom attributes
                            var fields = type.GetRuntimeFields();
                            foreach (var field in fields)
                            {
                                var attribute = (ParameterAttribute)field.GetCustomAttribute(typeof(ParameterAttribute));
                                if (attribute != null)
                                {
                                    if (attribute.Value.Equals(value))
                                        return field.GetValue(null);
                                }
                            }
                        }
                        // remove '-' from values coming in to be able to enum utf-8
                        stringValue = RemoveHyphenAndUnderscore(stringValue);
                        return Enum.Parse(type, stringValue, ignoreCase: true);
                    }
```